core/src/com/unciv/ui/screens/diplomacyscreen/DiplomacyScreen.kt

```diff
@@ -282,20 +282,22 @@ class DiplomacyScreen(
             && otherCivDiploManager.diplomaticStatus == DiplomaticStatus.DefensivePact
             && !otherCivDiploManager.otherCiv().isAtWarWith(viewingCiv) }
             .map { it.otherCiv() }.toMutableList()
-        // Go through and find all of the defensive pact chains and add them to the list
-        var listIndex = 0
-        while (listIndex < otherCivDefensivePactList.size) {
-            messageLines += if (viewingCiv.knows(otherCivDefensivePactList[listIndex]))
-                "[${otherCivDefensivePactList[listIndex].civName}] will also join them in the war"
-            else "An unknown civilization will also join them in the war"
-
-            // Add their defensive pact allies
-            otherCivDefensivePactList.addAll(otherCivDefensivePactList[listIndex].diplomacy.values
-                .filter { diploChain -> diploChain.diplomaticStatus == DiplomaticStatus.DefensivePact
-                    && !otherCivDefensivePactList.contains(diploChain.otherCiv())
-                    && diploChain.otherCiv() != viewingCiv && diploChain.otherCiv() != otherCiv
-                    && !diploChain.otherCiv().isAtWarWith(viewingCiv) }
-                .map { it.otherCiv() })
+            
+            // Defensive pact chains are not allowed now
+            var listIndex = 0
```

Comment:
Is there a reason to have this by index instead of 'for civ in othercovdefensivepactlist'?
I know this was here before you, but if we can improve readability, why not :)

Reply:
Is there a reason we're doing indexing at all on a list and not using ``for ((index, civ) in otherCivDefensivePactList.withIndexed()``. Like, "withIndexed" is explicitly in the Kotlin library so we don't need to do while loops for this

Reply:
Correct, but now it is no longer modified, so we can move to the more readable format
We don't even use the index anymore, just the civ